include/mbedtls/config_psa.h

```diff
@@ -593,8 +593,10 @@ extern "C" {
 #define MBEDTLS_PSA_BUILTIN_ALG_RSA_PSS 1
 #define PSA_WANT_ALG_RSA_PSS 1
 #endif /* MBEDTLS_PKCS1_V21 */
+#if defined(MBEDTLS_GENPRIME)
```

Comment:
This also disables signature and decryption.

Unfortunately, there is no way to express “private-key operations but not key generation” in the current PSA crypto configuration system. This is a known limitation. Resolving this limitation will require more design work.

Reply:
Maybe this is a bug, but I am able to perform a signature using an imported RSA key pair even when MBEDTLS_GENPRIME is not set (without this fix in Mbed TLS v2.24.0, and with this fix in v2.25.0). I can provide a sample application if that's helpful.

What's the intention behind both `MBEDTLS_PSA_BUILTIN_KEY_TYPE_RSA_KEY_PAIR ` and `MBEDTLS_PSA_BUILTIN_KEY_TYPE_RSA_PUBLIC_KEY `? Why differentiate?

The fix here provides symmetry in what `MBEDTLS_PSA_BUILTIN_KEY_TYPE_RSA_KEY_PAIR` sets, and what TLS options enable it.

There was (in Mbed TLS v2.24.0) a way to express “private-key operations but not key generation” in the PSA crypto configuration system. Was it intentional to remove this in v2.25.0 and up?

Reply:
The intention between differentiating have-key-pair and have-public-key is to allow builds that have public-key operations only (typically: I want to verify some signatures, but I'm not going to sign anything), which can save significant code size if the implementation is structured for it: you don't need any blinding code, CRT parameter handling, etc. Mbed TLS is not yet structured for it, but we'll work on that in the coming months.

Private key operations may have worked accidentally even when only `PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY` was enabled and not `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR`, but we're progressively tightening down on the configuration, and also ramping up testing (https://github.com/ARMmbed/mbedtls/pull/4444 will add systematic negative testing).

> There was (in Mbed TLS v2.24.0) a way to express “private-key operations but not key generation” in the PSA crypto configuration system.

In the classic configuration system, yes. In the PSA configuration system, no. We didn't intend to remove anything, but we're aware that the new system doesn't exactly match the granularity of the old one.

Reply:
I should also highlight that I'm not using `MBEDTLS_PSA_CRYPTO_CONFIG` or `PSA_WANT_*`. Indeed, with `PSA_WANT_*` there is no way to express "private-key operations but not key generation".

I am using the "old-style configuration mechanism", which it seems there is some intention to provide backwards compatibility with, hence my raising of this PR to help with that effort. If this aspect of PSA configuration isn't one you want to support because `PSA_WANT_*` can't express it yet, that's not great for my use case, but I can understand.

Another question would be if the configuration settings I've added to all.sh are considered invalid somehow. Is building with the default config minus `MBEDTLS_GENPRIME` (and no other changes) no longer supported?

Reply:
Breaking configurations without `MBEDTLS_GENPRIME` was an unintended regression. I fully agree that we should fix it. But I don't want to fix it by breaking something else. I think it should be possible to fix it by adding (back?) a few `#ifdef MBDETLS_GENPRIME` in the right places.

With `MBEDTLS_PSA_CRYPTO_CONFIG`, `MBEDTLS_GENPRIME` is a mandatory consequence of `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR` for the time being, that's a limitation to fix later.

Reply:
As a quick fix, I think the best solution would be to add `#if defined(MBEDTLS_GENPRIME)` back into `psa_crypto_*.c`. That's not the long-term solution, because it only matters for the software implementation, it doesn't allow e.g. an accelerator to declare that it can accelerate RSA signatures but not RSA key generation. But it's a good short-term solution because it fixes the problem you care about without breaking (or worsening) things that currently work (-ish), and it doesn't require any new design that we're not sure is the right design.